packages/amplify_api/lib/src/graphql/graphql_request_factory.dart

```diff
@@ -130,31 +143,34 @@ class GraphQLRequestFactory {
       String? id,
       required GraphQLRequestType requestType,
       required GraphQLRequestOperation requestOperation}) {
+    // retrieve schema from ModelType and validate required properties
     ModelSchema schema = _getAndValidateSchema(modelType, requestOperation);
 
-    // e.g. "Blog"
-    String name = schema.name;
-    // fields to retrieve, e.g. "id name createdAt"
-    String fields = _getFieldsFromModelType(schema);
+    // e.g. "Blog" or "Blogs"
+    String name = _getName(schema, requestOperation);
     // e.g. "query"
     String requestTypeVal = describeEnum(requestType);
     // e.g. "get"
     String requestOperationVal = describeEnum(requestOperation);
-    // {upper: "($id: ID!)", lower: "(id: $id)"}
-    DocumentInputs docInputs = _buildDocumentInputs(schema, requestOperation);
-
-    if (requestOperation == GraphQLRequestOperation.list) {
-      name = schema.pluralName!;
-      fields = 'items { $fields } nextToken';
-    }
-
-    String doc =
-        '''$requestTypeVal $requestOperationVal$name${docInputs.upper} { $requestOperationVal$name${docInputs.lower} { $fields } }''';
+    // e.g. {upper: "($id: ID!)", lower: "(id: $id)"}
+    DocumentInputs documentInputs =
+        _buildDocumentInputs(schema, requestOperation);
+    // e.g. "id name createdAt" - fields to retrieve
+    String fields = _getFieldsFromModelType(schema, requestOperation);
+    // e.g. "getBlog"
+    String requestName = "$requestOperationVal$name";
+    // e.g. query getBlog($id: ID!, $content: String) { getBlog(id: $id, content: $content) { id name createdAt } }
+    String document =
+        '''$requestTypeVal $requestOperationVal$name${documentInputs.upper} { $requestOperationVal$name${documentInputs.lower} { $fields } }''';
 
     // TODO: convert model to variable input for non-get operations
     Map<String, dynamic> variables =
         requestOperation == GraphQLRequestOperation.get ? {"id": id} : {};
 
-    return GraphQLRequest<T>(document: doc, variables: variables);
+    return GraphQLRequest<T>(
+        document: document,
+        variables: variables,
+        modelType: modelType,
+        decodePath: "$requestName");
```

Comment:
Since `requestName` is already a String, you can just write `decodePath: requestName`.